src/controllers/expenses.js

```diff
@@ -0,0 +1,130 @@
+/* eslint-disable space-before-function-paren */
+'use strict';
+
+const { getUserById } = require('../services/users');
+const {
+  getAll,
+  getOne,
+  create,
+  remove,
+  update,
+  normalize,
+} = require('../services/expenses');
+
+const getAllExpenses = async (req, res) => {
+  const { userId, from, to, categories } = req.query;
+  let preparedExpenses = await getAll();
+
+  if (userId) {
+    preparedExpenses = preparedExpenses
+      .filter(expense => expense.userId === +userId);
+  }
```

Comment:
In 20 line you can avoid converting `userId` several times for each iteration. You can make separate initialization for `userId` and make it number instantly 
`const userId = +req.query.userId`

Reply:
It's ok, but also you could use sequelize to filter your models but not array methods
If you are interested look through [this](https://sequelize.org/docs/v6/core-concepts/model-querying-basics/#applying-where-clauses)